test/e2e/validation_test.go

```diff
@@ -464,6 +465,107 @@ func testExpectedServicesRunning(t *testing.T) {
 	}
 }
 
+// testServicesConfigMap tests multiple aspects of expected functionality for the services ConfigMap
+// 1. It exists on operator startup 2. It is re-created when deleted 3. It is recreated if invalid contents are detected
+func testServicesConfigMap(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	operatorVersion, err := getWMCOVersion()
+	require.NoError(t, err)
+	// There is no easy way to populate ldflags when running e2e tests without having to maintain the WMCO version in
+	// another location, so that version.Get() will return something other than "". Because of this,
+	// we cannot use servicescm.Name and must build to the services CM name by retrieving the version in another manner.
+	servicesConfigMapName := servicescm.NamePrefix + operatorVersion
+
+	// Ensure the windows-services ConfigMap exists in the cluster
+	_, err = tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Get(context.TODO(),
+		servicesConfigMapName, meta.GetOptions{})
+	require.NoErrorf(t, err, "error retrieving ConfigMap: %s", servicesConfigMapName)
+
+	err = tc.testServicesCMRecreation(servicesConfigMapName)
+	require.NoErrorf(t, err, "error ensuring ConfigMap %s is re-created when deleted", servicesConfigMapName)
+
+	err = tc.testInvalidServicesCM(servicesConfigMapName)
+	require.NoError(t, err, "error testing handling of invalid ConfigMap")
+}
+
+// testServicesCMRecreation tests that if the services ConfigMap is deleted, a valid one is re-created in its place
+func (tc *testContext) testServicesCMRecreation(cmName string) error {
+	if err := tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Delete(context.TODO(), cmName, meta.DeleteOptions{}); err != nil {
+		return err
+	}
```

Comment:
Remind me, is this going to return only after the object is fully deleted?
Want to make sure we aren't getting the same object that is actively being deleted

Reply:
I'm pretty sure it returns after getting a response from the server, in the code I see no guarantee it waits for the resource has been fully deleted. However, that's the reasoning behind `waitForValidWindowsServicesConfigMap()` - if for some reason we get the same object that should be deleted, [the test will not pass until we get a valid (re-created) one](https://github.com/saifshaikh48/windows-machine-config-operator/blob/maintain-services-configmap/test/e2e/validation_test.go#L558-L561).